src/styles/shared/icons.scss

```diff
@@ -2,8 +2,16 @@
   @return rem(20px);
 }
 
-@mixin recolor-icon($main-color, $secondary-color: null) {
-  fill: $main-color;
+@mixin recolor-icon($fill-color, $secondary-color: null, $filter-color: null) {
+  svg {
+    fill: $fill-color;
+  }
+
+  @if $filter-color != null {
+    img {
+      filter: $filter-color;
```

Comment:
Looking at the docs for `filter` (https://developer.mozilla.org/en-US/docs/Web/CSS/filter) it doesn't look like you can just pass in a color string. How does this work?

Reply:
There's a map of filter values that match with the polaris colors from `polaris-tokens`: https://github.com/Shopify/polaris-tokens/blob/master/dist/color-filters.json

In a previous PR, I added the filter scss function that allows us to use grab the values using the same naming convention.

Reply:
Oh I see, I didn't realize the values were coming from that file. This code makes sense then, and I'd rather use the Icon component to render icons than spans so this is good IMO

Reply:
The `$filter-color` variable is the value that comes back from the `filter` function. The filter function handles the validation of the input to ensure it is the correct color. So this variable would be something like:

 `brightness(0) saturate(100%) invert(28%) sepia(83%) saturate(3919%) hue-rotate(168deg) brightness(93%) contrast(101%)`

Reply:
Derp, sorry I missed that! You’re absolutely right 👍